### PR TITLE
[MPS] Add lerp implementation

### LINKED ISSUE #105382: Need support and testing for Adam optimizer for MPS
### 🚀 The feature, motivation and pitch

Environment of Mac M2
```
Python3.10
torch              2.1.0.dev20230717
torchaudio         2.1.0.dev20230717
torchvision        0.15.2a0
```

I want to use the Adam optimizer to train my model. And got an error:
```
NotImplementedError: The operator 'aten::lerp.Scalar_out' is not currently implemented for the MPS device. If you want this op to be added in priority during the prototype phase of this feature, please comment on https://github.com/pytorch/pytorch/issues/77764. As a temporary fix, you can set the environment variable `PYTORCH_ENABLE_MPS_FALLBACK=1` to use the CPU as a fallback for this op. WARNING: this will be slower than running natively on MPS
```
And when I set the `PYTORCH_ENABLE_MPS_FALLBACK=1`, the training speed is quite slower.

I'm testing the [tiny VIT model](https://github.com/UdbhavPrasad072300/Transformer-Implementations/blob/main/transformer_package/models/transformer.py#L406) with minist dataset.

The details is following:
M2 chip takes about **2.4 minutes** on CPU with Adam for one epoch.
M2 chip takes about **2.0 minutes** on GPU with Adam for one epoch( with `PYTORCH_ENABLE_MPS_FALLBACK=1`).
M2 chip takes about **30 seconds** on GPU with SGD for one epoch.

### Alternatives

_No response_

### Additional context

_No response_

cc @vincentqb @jbschlosser @albanD @janeyx99 @crcrpar @kulinseth @malfet @DenisVieriu97 @razarmehr @abhudev @ezyang @gchanan @zou3519

### PR DESCRIPTION
**Pull request by malfet**
lerp.Scalar fits very well into binary op template
Add a very naive implementation for `lerp.Tensor` as `add_out(self, weights.mul(end.sub(self)))`

Enable `lerp` testing in `test_mps`

Fixes https://github.com/pytorch/pytorch/issues/105382

### CODE REVIEW

File: aten/src/ATen/native/mps/operations/BinaryOps.mm [hunk 1]
```diff
@@ -273,18 +274,31 @@ void add_sub_template(const Tensor& self,
     at::native::alpha_check(commonDtype, alpha);
   }
 
-  BinaryOpBlock add_sub_op_block = ^BinaryOpFn(cachedGraph, primaryCastTensor, secondaryCastTensor) {
+  if (!alpha_has_value && op_name == "lerp") {
+    if (!self.is_alias_of(other)) { // if inplace, no-op
+      const_cast<Tensor&>(output) = other.clone();
```
[the remark this answers is not in the document]
Comment (malfet) [reply aten/src/ATen/native/mps/operations/BinaryOps.mm:279]: I was about to ask that question. Do you know the right semantic for `at::add_out`(which requires the same ugly `const_cast`)

[the remark this answers is not in the document]
Comment (albanD) [reply aten/src/ATen/native/mps/operations/BinaryOps.mm:279]: The `TensorImpl` form the output should be preserved. You are allowed to `resize_()` it as needed. But then you need to fill it.
If there is enough space there already, you must use it as-is.

**Review by albanD — APPROVED** [review]
Const correctness of the structured kernels seems double broken :(
But sounds good to me.

**Pull request closed without merging**